File: C++/GCD.cpp
//gcd or hcf
#include<bits/stdc++.h>
using namespace std;

int gcd(int a, int b) {
	if (b == 0) {
		return a;
	}
	return gcd(b, a % b);
}

int main()
{
    //*************//*brute force
    int ans=0,a=5,b=10;
    for(int i=1;i<=min(a,b);i++)
{
    if(a%i == 0 && b%i == 0)
    ans=i;
}
cout<<ans;
/**************************/

	int a = 4, b = 8;
	cout <<"The GCD of the two numbers is "<<gcd(a, b);
}

//*************************************************************

File: gcd.java
// Java program to find GCD of two numbers
public class GFG {
     
    // Function to return gcd of a and b
    static int gcd(int a, int b)
    {
        int result = Math.min(a, b); // Find Minimum of a nd b
        while (result > 0) {
            if (a % result == 0 && b % result == 0) {
                break;
            }
            result--;
        }
        return result; // return gcd of a nd b
    }
     
    // Driver program to test above function
    public static void main (String[] args)
    {
        int a = 98, b = 56;
        System.out.print("GCD of " + a + " and " + b + " is " + gcd(a, b));
    }
}
